dataprofiler/labelers/column_name_model.py

```diff
@@ -0,0 +1,253 @@
+"""Contains class for column name data labeling model."""
+import json
+import os
+
+import numpy as np
+
+try:
+    import rapidfuzz
+except ImportError:
+    # not required when using `ColumnNameModel`, but
+    # will below recommend to install if not installed'
+    # when running `predict()`
+    pass
+
+from .. import dp_logging
+from ..reports.utils import require_module
+from .base_model import AutoSubRegistrationMeta, BaseModel
+
+logger = dp_logging.get_child_logger(__name__)
+
+
+class ColumnNameModel(BaseModel, metaclass=AutoSubRegistrationMeta):
+    """Class for column name data labeling model."""
+
+    def __init__(self, parameters=None):
+        """Initialize function for ColumnNameModel.
+
+        :param parameters: Contains all the appropriate parameters for the model.
+            Possible parameters are:
+                max_length, max_num_chars, dim_embed
+        :type parameters: dict
+        :return: None
+        """
+        # parameter initialization
+        if not parameters:
+            parameters = {}
+        parameters.setdefault("false_positive_dict", None)
+        parameters.setdefault("true_positive_dict", None)
+
+        # initialize class
+        self._validate_parameters(parameters)
+        self._parameters = parameters
+
+    def _validate_parameters(self, parameters):
+        r"""
+        Validate the parameters sent in.
+
+        Raise error if invalid parameters are present.
+
+        :param parameters: parameter dict containing the following parameters:
+            true_positive_dict
+            false_positive_dict
+        :type parameters: dict
+        :return: None
+        """
+        errors = []
+
+        list_of_accepted_parameters = [
+            "true_positive_dict",
+            "false_positive_dict",
+        ]
+
+        for param in parameters:
+            value = parameters[param]
+            if (
+                param == "false_positive_dict"
+                and value is not None
+                and (not isinstance(value, list) or "attribute" not in value[0].keys())
+            ):
+                errors.append(
+                    """`{}` must be a list of dictionaries with at
+                    least 'attribute' as the key""".format(
+                        param
+                    )
+                )
+            elif param == "true_positive_dict" and (
+                not isinstance(value, list) or not isinstance(value[0], dict)
+            ):
+                errors.append(
+                    """`{}` must be a list of dictionaries each with the following
```

Comment:
doesn't check for `attribute` and `label` key in list of dicts. Does it need to?

Reply:
fixing -- good catch